plugin/src/main/groovy/com/novoda/staticanalysis/internal/VariantAware.groovy

```diff
@@ -0,0 +1,37 @@
+package com.novoda.staticanalysis.internal
+
+import org.gradle.api.DomainObjectSet
+import org.gradle.api.NamedDomainObjectSet
+
+trait VariantAware {
```

Comment:
Personally I would haven chosen composition over inheritance here since we mostly expose fields and methods to be shared.